### PR TITLE
Consumer, Producer, TopicPartition classes are now sub-classable

### PR DESCRIPTION
**Pull request by edenhill**

### CODE REVIEW

File: confluent_kafka/src/Consumer.c [hunk 1]
```diff
@@ -55,7 +55,7 @@ static void Consumer_dealloc (Handle *self) {
 	if (self->rk)
 		rd_kafka_destroy(self->rk);
 
-	Py_TYPE(self)->tp_free((PyObject *)self);
+        Py_TYPE(self)->tp_free((PyObject *)self);
```
Comment (ewencp) [inline confluent_kafka/src/Consumer.c]: We seem to be getting mixed tabs & spaces...

Comment (edenhill) [inline confluent_kafka/src/Consumer.c]: The file originally has tabs for indention, which sucks but I probably coded it with the wrong .emacs file in place.
New changes use space for indent instead.
We can do a forklift untabify at some point.

**Comment by edenhill** [comment]
@roopahc and @criccomini  I'd like you to try this out before I merge it.

**Comment by roopahc** [comment]
@edenhill @ewencp I tested with AvroProducer class. It works good.. Thanks for this update.

**Comment by edenhill** [comment]
@roopahc Thanks!

**Pull request merged**